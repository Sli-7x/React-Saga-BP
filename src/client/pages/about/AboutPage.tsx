import * as React from 'react';
import { Helmet } from 'react-helmet';
import { withError } from '../../components/errors/ErrorWrapper';
import { Button } from '../../components/ui/button/Button';

export class AboutPage extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.onClick = this.onClick.bind(this);
    this.state = {
      show: false,
    };
  }

  onClick() {
    this.setState({ show: true }, () => {
      throw new Error('asd');
    });
  }

  render() {
    return (
      <div>
        <Helmet>
          <title>Typescript react ssr | About</title>
        </Helmet>
        ABOUT US
        <Button onClick={this.onClick}>Click me</Button>
      </div>
    );
  }
}

export default withError(AboutPage);
