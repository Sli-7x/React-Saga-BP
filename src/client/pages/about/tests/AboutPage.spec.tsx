import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { AboutPage } from '../AboutPage';
import { shallow } from 'enzyme';

describe('AboutPage.tsx', () => {
  it('should render correctly', () => {
    const filledTree = renderer.create(<AboutPage />).toJSON();
    expect(filledTree).toMatchSnapshot();
  });

  it('should cover onClick()', () => {
    const component = shallow(<AboutPage />);
    component.setState = jest.fn();
    const instance = component.instance() as AboutPage;
    instance.onClick();
    expect(component.setState).toHaveBeenCalled();
  });
});
