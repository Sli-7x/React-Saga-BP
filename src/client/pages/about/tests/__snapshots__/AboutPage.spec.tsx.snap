// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage.tsx should cover onClick() 1`] = `
<div>
  ABOUT US
  <button
    className="sc-bdVaJa cqFNto"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;

exports[`AboutPage.tsx should render correctly 1`] = `
<div>
  ABOUT US
  <button
    className="sc-bdVaJa cqFNto"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
